tests/unit/test_regridding_service.py

```diff
@@ -1,1089 +1,1498 @@
 """Tests regridding service."""
 
 import re
-from logging import getLogger
 from pathlib import Path
-from shutil import rmtree
-from tempfile import mkdtemp
-from unittest import TestCase
-from unittest.mock import patch
+from unittest.mock import MagicMock, patch
 from uuid import uuid4
 
 import numpy as np
-from harmony_service_lib.message import Message
-from netCDF4 import Dataset, Variable
+import pytest
+import xarray as xr
+from harmony_service_lib.message import Message as HarmonyMessage
+from harmony_service_lib.message import Source as HarmonySource
+from netCDF4 import Dataset, Dimension
 from numpy.testing import assert_array_equal
+from pyproj import CRS
 from pyresample.geometry import AreaDefinition
 from varinfo import VarInfoFromNetCDF4
 
-import harmony_regridding_service.regridding_service as rs
 from harmony_regridding_service.exceptions import (
+    InvalidSourceCRS,
     InvalidSourceDimensions,
     RegridderException,
+    SourceDataError,
+)
+from harmony_regridding_service.regridding_service import (
```

Comment:
Possibly missing tests for:

* `_compute_area_extent_from_regular_x_y_coords`
* `_compute_source_swath`
* `_cache_resamplers` (This naively looks like thing that is a bit more difficult to test, with potentially a bunch of mocks involved, so I'm guessing that's why)

Reply:
_compute_area_extent_from_regular_x_y_coords:  [`04ce080` (#22)](https://github.com/nasa/harmony-regridding-service/pull/22/commits/04ce0809af16ad01f1046229a4937e206deca509)

Reply:
Discussed on call - totally fair that these tests are descending into the realm of heavy `Mock` usage. And also totally fine with further test coverage being added as nearby updates are made. (I think this PR is just victim of having a big diff making me look at more of the code)

Reply:
also _compute_source_swath 88a5430